### ISSUE TITLE
Sources / GraphQL

### ISSUE THREAD

**Issue by coreyward** [wiziple/gatsby-starter-default-intl#3]
This would be more useful if it was adapted to work with source plugins (e.g. `gatsby-source-contentful`) rather than relying on outdated, hardcoded translation files. It seems unrealistic that a majority of Gatsby websites would forgo GraphQL querying in order to support translations.

**Comment by jasonmit** [comment]
I rewrote parts of this to work with source plugins.

It was a bit of a challenge in getting the translations out of the graphql API.  We're limited to the using only Page queries since Pages are the only parts of a Gatsby app that can make GraphQL queries with variables in them.  The variable we would be querying off of us the locale/lang coming from the pageContext.

I'm hoping `<StaticQuery>` is updated to handle variables - which would then solve this and I can package up a solution.

**Comment by benknight** [comment]
Agreed, I source my translations from Airtable so I can't use this i18n plugin.

**Comment by mikexavier** [comment]
Hey @jasonmit do you have an example of passing pageContext.intl.lang to a page query?

I am using WordPress and Polylang and the query looks something like this:

```
export const query = graphql`
  query($lang: WP_LanguageCodeEnum = EN) {
    wp {
      page(id: "5", idType: DATABASE_ID) {
        translation(language: $lang) {
          homeBeta {
            introHeader
          }
        }
      }
    }
  }
`
```

I would like it so that when a user clicks to "changeLocale" it updates that variable from EN to DE etc.

I'm new to this, so have just been experimenting... I'm not sure if `$lang: WP_LanguageCodeEnum = EN ` is the right way of going about this. Happy to learn, if you could point me In the right direction, I'd appreciate it :) thx
